### PR TITLE
elfwriter-test: Clean the files

### PR DESCRIPTION
**Pull request by v-thakkar**
Signed-off-by: Vaishali Thakkar <me.vaishalithakkar@gmail.com>

### CODE REVIEW

File: pkg/elfwriter/elfwriter_test.go [hunk 1]
```diff
@@ -265,7 +265,7 @@ func TestWriter_HasLinks(t *testing.T) {
 	inElf, err := elf.Open("testdata/libc.so.6")
 	require.NoError(t, err)
 	t.Cleanup(func() {
-		inElf.Close()
+		defer inElf.Close()
```
Comment (kakkoyun) [inline pkg/elfwriter/elfwriter_test.go:268]:
```suggestion
	inElf.Close()
```

You don't need to `defer` in here.

**Review by kakkoyun — CHANGES_REQUESTED** [review]

File: pkg/elfwriter/elfwriter_test.go (hunk 1 above)
Comment (v-thakkar) [reply pkg/elfwriter/elfwriter_test.go]: Ok, I see that it's part of idiomatic Go. Thanks. Removed it at one other place in the file too.

Comment (kakkoyun) [reply pkg/elfwriter/elfwriter_test.go]: 👍  It's not about being idiomatic here. First of all, there are no other statements in that scope (the scope is the closure here). And `t.Cleanup` actually does the trick of the `defer.` It gets to run after the test exists.

**Review by kakkoyun — APPROVED** [review]

File: pkg/elfwriter/elfwriter_test.go (hunk 1 above)
Comment (v-thakkar) [reply pkg/elfwriter/elfwriter_test.go]: Ah, yes. I see what you're saying now. :+1:

**Review by javierhonduco — COMMENTED** [review]
Thanks for doing this!!

**Pull request merged**